### PR TITLE
Build fixes for C++17, Apple Silicon

### PR DESCRIPTION
**Pull request by dcwhite**
Not quite ready for review, but leaving this here to run all the builds.

I might get it ready over my ✈️ 🛄 🍹 

Closes https://github.com/SCIInstitute/SCIRun/issues/2392

### CODE REVIEW

**Comment by dcwhite** [comment]
Doesn't build on 10.14

File: src/Core/Logging/Log.cc [hunk 1]
```diff
@@ -94,8 +94,8 @@ namespace
   protected:
     void sink_it_(const spdlog::details::log_msg& msg) override
     {
-		  fmt::memory_buffer formatted;
-		  sink::formatter_->format(msg, formatted);
+		  spdlog::memory_buf_t formatted;
```
Comment (jessdtate) [inline src/Core/Logging/Log.cc:97]: I get some build errors with this:
```
/Users/jess/software/SCIRun_testing/src/Core/Logging/Log.cc:97:26: error: use of undeclared identifier 'formatted'; did you
      mean 'fmt::formatter'?
                  spdlog::memory_buf_t formatted;
                                       ^~~~~~~~~
                                       fmt::formatter
/Users/jess/software/SCIRun_testing/bin/Externals/Source/SpdLog_external/include/spdlog/fmt/bundled/core.h:491:8: note: 
      'fmt::formatter' declared here
struct formatter {
       ^
/Users/jess/software/SCIRun_testing/src/Core/Logging/Log.cc:97:5: error: no type named 'memory_buf_t' in namespace 'spdlog';
      did you mean 'fmt::memory_buffer'?
                  spdlog::memory_buf_t formatted;
                  ^~~~~~~~~~~~~~~~~~~~
                  fmt::memory_buffer
/Users/jess/software/SCIRun_testing/bin/Externals/Source/SpdLog_external/include/spdlog/fmt/bundled/format.h:553:35: note: 
      'fmt::memory_buffer' declared here
typedef basic_memory_buffer<char> memory_buffer;
```

Comment (dcwhite) [reply src/Core/Logging/Log.cc:97]: With two externals updated, you'll need to do a fresh Superbuild. Try blowing away the whole directory.

**Comment by dcwhite** [comment]
Just in time for a new version of qt-install-action! Looks like it's working again

**Review by jessdtate — APPROVED** [review]

**Review by jessdtate — APPROVED** [review]

**Pull request merged**